pkg/build/builder/source.go

```diff
@@ -73,7 +73,7 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 func checkRemoteGit(url string, timeout time.Duration) error {
 	glog.V(4).Infof("git ls-remote --heads %q", url)
 	cmd := exec.Command("git", "ls-remote", "--heads", url)
-	cmd.Env = []string{"GIT_ASKPASS=/bin/true"}
+	cmd.Env = []string{"GIT_ASKPASS=/bin/true", "GIT_SSH="+os.Getenv("GIT_SSH")}
```

Comment:
does this cause problems in cases where the url is not an SSH url and GIT_SSH is going to end up being an empty string?